server/container/util/src/main/java/org/apache/james/util/concurrency/ConcurrentTestRunner.java

```diff
@@ -38,19 +38,28 @@
 
     public static final int DEFAULT_OPERATION_COUNT = 1;
 
+    @FunctionalInterface
+    public interface RequireOperation {
+        RequireThreadCount operation(BiConsumer operation);
+    }
+
+    @FunctionalInterface
+    public interface RequireThreadCount {
+        Builder threadCount(int threadCount);
+    }
+
     public static class Builder {
-        private Optional<Integer> threadCount;
+        private final int threadCount;
+        private final BiConsumer operation;
```

Comment:
you don't have warnings about generics ?

Reply:
Ok, it's not really a Biconsumer, you reused a name from java.function, It's not a very good idea.
I think that renaming it ConcurrentOperation is much better naming